### PR TITLE
(RFC) Show now handles `;` in a tuple

### LINKED ISSUE #32775: improve show to handle `;` in a tuple
```
julia> ( :((a,; b)) ) # :/ would be nice for this to round-trip through `show`
:(($(Expr(:parameters, :b)), a))

julia> :( g(a,; b) ) # should print like this, using the same code as handles `:call`:
:(g(a; b))
```

### PR DESCRIPTION
**Pull request by lhnguyen-vn**
Fix #32775

### CODE REVIEW

File: base/show.jl [hunk 1]
```diff
@@ -1168,7 +1168,13 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int)
             sep = ", "
         end
         head !== :row && print(io, op)
-        show_list(io, args, sep, indent)
+        if isa(args[1], Expr) && args[1].head === :parameters
```
Comment (c42f) [inline base/show.jl]: You can use `Meta.isexpr(args[1], :parameters)` for this.

**Review by c42f — COMMENTED** [review]
Cool, generally seems useful to have more round-trippable syntax :+1:

It seems there's an out of bounds bug somewhere here which is causing CI failures. Might be because `args` can be length 0?

**Comment by lhnguyen-vn** [comment]
Thanks for the response! I totally forgot about empty tuples, it has been updated now.

**Comment by c42f** [comment]
Oh, it appears that `Meta.isexpr` is not available during bootstrap which has caused CI to fail (even though it's used elsewhere in this file, those code paths must not hit during bootstrap).

The easiest fix is to go back to using the explicit version without `Meta.isexpr`... sorry!

File: test/show.jl [hunk 2]
```diff
@@ -753,6 +753,9 @@ test_mt(show_f5, "show_f5(A::AbstractArray{T,N}, indices::Vararg{$Int,N})")
 @test_repr "[a;]"
 @test_repr "[a; b]"
 
+# Issue #32775, printing ; in a tuple
+@test sprint(show, :((a,;b))) == ":((a; b))"
```
Comment (JeffBezanson) [inline test/show.jl]: This printing is incorrect; `(a,;b)` and `(a; b)` are different (the comma makes it a tuple).

Comment (lhnguyen-vn) [reply test/show.jl]: Maybe I'm missing something, but wouldn't Julia parse `(a; b)` as a block?

**Comment by JeffBezanson** [comment]
Taken care of by #34038

**Pull request closed without merging**